docs/ides/configuring-emacs-tramp.md

```diff
@@ -0,0 +1,56 @@
+# Configuring Emacs TRAMP
+[Emacs TRAMP](https://www.emacswiki.org/emacs/TrampMode) is a method of running editing operations on a remote server.
+
+## Connecting To A Workspace
+To connect to your workspace first run:
+
+```
+coder config-ssh
+```
+
+Then you can connect to your workspace by its name in the format: `coder.<name>`.
+
+In Emacs type `C-x d` and then input: `/ssh:coder.<name>:` and hit enter. This will open up Dired on the workspace's home directory.
+
+## Directory Tracking
+### `ansi-term`
+If you run your terminal in Emacs via `ansi-term` then you might run into a problem where while SSH-ed into a workspace Emacs will not change its `default-directory` to open files in the directory your shell is in.
+
+To fix this:
+
+1. In your Emacs `init.el` file add:
+   ```lisp
+   (setq tramp-default-method "ssh")
+   ```
+2. Then on your Coder workspace instance be sure to set the hostname to the `coder.<name>` format:
+   ```bash
+   hostname coder.<name>
+   ```
+   This can also be done in the workspace Terraform template by setting workspace instance's hostname to the data `coder_workspace.name` attribute. How this is done depends on how the instance is provisioned.
```

Comment:
Could we change `ansi_term_announce_host` to `printf '\033AnSiTh %s\n' "coder.$(hostname)"` instead?  That seems like it might be an easier change and will work regardless of the provisioning method.

Reply:
I think maybe I need to rewrite this section to say:

> In your workspace Terraform template be sure to add the following
> ```hcl
> data "coder_workspace" "me" {
> }
>
> resource "coder_agent" "main" {
>   # ...
>   env {
>     name = "CODER_WORKSPACE_NAME"
>     value = data.coder_workspace.me.name
>   }
> }
> ```

And then modify the `ansi_term_announce_host` function to do `printf '\033AnSiTh %s\n' "coder.$CODER_WORKSPACE_NAME"`

At least on AWS `coder.$(hostname)` by default would either result in `coder.ip-xxx-xx-x-xxx` or `coder.coder-user-user` depending on the hostname scheme you configure.

Reply:
Ahhhhh OK that seems reasonable.

Reply:
Looks like we do a similar thing with our own template: https://github.com/coder/coder/blob/a37e61a099c2dcb65a41e28cecd8aba126eb5d3e/dogfood/main.tf#L58